ledger/archival_test.go

```diff
@@ -132,6 +147,134 @@ func TestArchival(t *testing.T) {
 	t.Error("Did not observe every tracker GCing the ledger")
 }
 
+func TestArchivalRestart(t *testing.T) {
+	// Start in archival mode, add 2K blocks, restart, ensure all blocks are there
+
+	dbTempDir, err := ioutil.TempDir("", "testdir"+t.Name())
+	require.NoError(t, err)
+	dbName := fmt.Sprintf("%s.%d", t.Name(), crypto.RandUint64())
+	dbPrefix := filepath.Join(dbTempDir, dbName)
+	defer os.RemoveAll(dbTempDir)
+
+	seed := getInitState()
+	const inMem = false // use persistent storage
+	const archival = true
+
+	l, err := OpenLedger(logging.Base(), dbPrefix, inMem, seed, archival)
+	require.NoError(t, err)
+	blk := seed.InitBlocks[0]
+
+	const maxBlocks = 2000
+	for i := 0; i < maxBlocks; i++ {
+		blk.BlockHeader.Round++
+		blk.BlockHeader.TimeStamp += int64(crypto.RandUint64() % 100 * 1000)
+		l.AddBlock(blk, agreement.Certificate{})
+
+		// Don't bother checking the trackers every round -- it's too slow..
+		if crypto.RandUint64()%23 > 0 {
```

Comment:
Can we avoid using randomness here ? rationale is that we want our tests to be repetitive ( if we can ).

Reply:
Removed the check, not needed anymore